### PR TITLE
WIP: Source for refactoring patches out of #244

### PR DESCRIPTION
**Pull request by shonfeder**
This PR is just to bring visibility to the branch being used to track the refactoring of #244 into smaller change sets.

The branch should shrink over time, as pieces are broken out and merged into unstable.

### CODE REVIEW

File: tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaFunOper.scala [hunk 1]
```diff
@@ -103,27 +105,29 @@ object TlaFunOper {
     *
     * <p>
     * `TlaOperDecl("Fact", List(),
-    *   OperEx(recFunDef,
```
Comment (shonfeder) [inline tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaFunOper.scala:106]: @konnov do want to include these formatting changes to the doc comments, or was this inadvertent?

Comment (konnov) [reply tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaFunOper.scala:106]: Let's skip this change. This must have been automatically reformatted

**Comment by shonfeder** [comment]
Finished! After pushing the latest updates, this was automatically closed, since it's now identical with unstable. :tada: 

Now I'm gonna touch up notes on the refactoring process, and we'll see where those might belong :)

**Comment by konnov** [comment]
This is amazing. Thanks a lot @shonfeder!

**Pull request closed without merging**